code/src/Grid/AbstractGrid.ts

```diff
@@ -158,21 +170,22 @@ namespace Grid {
             return gridID === this._uniqueId || gridID === this._widgetId;
         }
 
-        /**
-         * Get the column on the grid by giving a columnID or a binding.
-         * @param key key can be a columnID or a binding of a column
-         * @returns Column with the same columnID or binding.
-         */
         public getColumn(key: string): Column.IColumn {
             if (this._columns.has(key)) {
                 return this._columns.get(key);
             } else {
-                return _.toArray(this.columns)
-                    .map((p) => p[1])
-                    .find((p) => p && p.equalsToID(key));
+                return this.getColumns().find((p) => p && p.equalsToID(key));
             }
         }
 
+        public getColumns(): Column.IColumn[] {
+            return Array.from(this._columnsSet);
+        }
+
+        public hasColumn(key: string): boolean {
+            return this._columns.has(key);
```

Comment:
Shouldn't this method, look for the other ID's as well? or should we keep this case for a future improvement?